src/System.Json/tests/JsonArrayTests.cs

```diff
@@ -0,0 +1,337 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.IO;
+using System.Text;
+using Xunit;
+using System.Collections.Generic;
+using System.Collections;
+
+namespace System.Json.Tests
+{
+    public class JsonArrayTests
+    {
+        public static IEnumerable<object[]> JsonValues_TestData()
+        {
+            yield return new object[] { new JsonValue[0] };
+            yield return new object[] { new JsonValue[] { null } };
+        }
+
+        [Theory]
+        [MemberData(nameof(JsonValues_TestData))]
+        public void Ctor(JsonValue[] items)
+        {
+            VerifyJsonArray(new JsonArray(items), items);
+            VerifyJsonArray(new JsonArray((IEnumerable<JsonValue>)items), items);
+        }
+
+        private static void VerifyJsonArray(JsonArray array, JsonValue[] values)
+        {
+            Assert.Equal(values.Length, array.Count);
+            for (int i = 0; i < values.Length; i++)
+            {
+                Assert.Equal(values[i], array[i]);
+            }
+        }
+        
+        [Fact]
+        public void Ctor_Array_Works()
+        {
+            JsonValue[] items = new JsonValue[] { new JsonPrimitive(true) };
+            JsonArray array = new JsonArray(items);
+            Assert.Equal(1, array.Count);
+            Assert.Same(items[0], array[0]);
+        }
+
+        [Fact]
+        public void Ctor_IEnumerable_Works()
+        {
+            JsonValue[] items = new JsonValue[] { new JsonPrimitive(true) };
```

Comment:
Ditto; if this is added to JsonValues_TestData, you don't need either of these tests, right?